### DISCUSSION TITLE
Where/how to use a right_margin inside the graph with the new deprecation

### DISCUSSION THREAD

**Discussion by r-jean-pierre** [dbuezas/lovelace-plotly-graph-card#218] [Q&A] [answered]
Hi,

I just discover the new deprecation with the combo:
```yaml
show_value:
   right_margin: 12
```
but I didn't really understand how to add a gap **inside** the graph to let some space to display the value:

![2023-01-30 21_10_24-Window](https://user-images.githubusercontent.com/9727894/215584683-eb9c319d-048f-4157-bbda-8b3abf914e04.png)

if you look carefully, I have a second y-axis, and I was wondering if the tricks in the doc will really work in that case

Please note that things like:
```yaml
layout:
  margin:
     r: 32
````
Is not what I'm looking for as it will just create an unnecessary large empty margin between the graph and the border of the card and won't solve the collision showvalue with 2nd y-axis ticks

Also, as it was the case before, but i was to shy to ask,  how to enter this gap width in term of pixels rather than in %, as a percentage of the x-axis will change the the precise location of the text itself after a zoom in/out.

**Comment by dbuezas** [comment, marked as answer]
You can use time_offset as the error message indicates. There's also some more help (new) at the bottom of the readme.
The old right_margin created a bunch of issues, and it is redundant.
Much luck with this!

**Comment by dbuezas** [comment]
To do it in %, you'll need to use a $fn on the global  time_offset and make it relative to hours_to_show

**Reply by r-jean-pierre** [reply]
Hi!

Thanks for the trick , but correct me if I'm wrong: it corrects from a pixel point of view, but then all data are timestamp shifted?

![2023_01_31_20_17_38_Window](https://user-images.githubusercontent.com/9727894/215861685-0fd88ddd-d440-45b2-96c3-ea517ffcd1bb.png)


On the example above, of course, reading the 13.9° is much more easy and fancy than the humidity percentage, but in that case, as I used **time_offset: '-6h'** just to visually adapt, as a consequence the data point is really shifted on the x-axis, and as you can see in the hover information, the timestamp of the point is shifted too.

Maybe I missed something in the way of creating a space on the right of the curve, but I'm not a big fan of shifting data values instead of just adding white pixels

Let's says there is few other options, can it be possible maybe to right-align the text itself and place it on top of the curve? In the worst case scenario the last datapoint itself will be close to the border, but the text above will also never go beyond the grid
Something like this:
![2023_01_31_20_31_56_Window](https://user-images.githubusercontent.com/9727894/215863811-564aa21e-52fe-469c-896f-8a53e3d53458.png)

**Reply by dbuezas** [reply]
I mean the global one, not inside the entity. That one just offsets the viewport, it doesn't touch the data

**Reply by r-jean-pierre** [reply]
Oh! Okay! It took me time to understand where and how! Now it sound obvious...  :-)

**Reply by dbuezas** [reply]
:) yes, In retrospective, I should have named it differently to the one inside entities. Global_offset or viewport_offset or something of the sort.

**Reply by dbuezas** [reply]
I should add some happy face emojis so ppl read my error messages, hehe
<img width="429" alt="image" src="https://user-images.githubusercontent.com/777196/215887317-50585a21-eb0f-424c-8148-99a8e23a5eb5.png">

**Reply by r-jean-pierre** [reply]
:)

Well in, fact from my side, the "global" confused me, I started to put 'time_offset' everywhere (layout, entities, etc. ) except really "outside" at the same level at hours_to_show or refresh_interval. On top of that I was using time_offset: -8h, with minus :) 
Well definitely I'm a casual user, any pixel tweak costs me hours of trial-and-error...

**Reply by dbuezas** [reply]
And the readme has become very unorganised. I hope a contributor dares at some point and improves it by converting it to a table or something like that.

**Comment by r-jean-pierre** [comment]
BTW here is where I am:
![2023-01-31 22_37_54-YAML Dashboard – Home Assistant](https://user-images.githubusercontent.com/9727894/215889532-c85319e8-5cfa-49ac-8524-141af4088019.png)
Still need some fine tuning about padding & spaces but globally, this is what I wanted to have, and I really enjoy the plotly interactivity + statistics.

**Reply by dbuezas** [reply]
Wow, very pretty! what are those gauges at the top?

**Reply by miljbee** [reply]
Would you mind posting a yaml, I would like to copy/paste your work ! I am interrested in having min/mean/max colored. Thx !

**Reply by r-jean-pierre** [reply]
> Wow, very pretty! what are those gauges at the top?

I used this CC: https://github.com/custom-cards/dual-gauge-card

**Reply by r-jean-pierre** [reply]
> Would you mind posting a yaml, I would like to copy/paste your work ! I am interrested in having min/mean/max colored. Thx !

Yes, you can find the skeleton there:

https://github.com/dbuezas/lovelace-plotly-graph-card/discussions/81

I posted the YAML at the end of that discussion
